apps/oc_chef_wm/src/oc_chef_wm_authn_ldap.erl

```diff
@@ -142,21 +162,30 @@ maybe_encrypt_session(start_tls, {ok, Session}, Timeout) ->
 maybe_encrypt_session(_, {ok, Session}, _) ->
     {ok, Session}.
 
-
+-spec canonical_username(string()) -> binary().
+canonical_username(Username) ->
+    characters_to_binary(
+      re:replace(string:to_lower(Username),
+                 "[^a-z0-9_-]",
+                 "_",
+                 [{return, list}, global])).
 
 result_to_user_ejson(_, UserName, []) ->
     lager:info("User ~p not found in LDAP", [UserName]),
     {error, unauthorized};
-result_to_user_ejson(LoginAttr, _, [{eldap_entry, CN, DataIn}|_]) ->
+result_to_user_ejson(LoginAttr, UserName, [{eldap_entry, CN, DataIn}|_]) ->
 
     % No guarantees on casing, so let's not make assumptions:
     Data = [ { string:to_lower(Key), Value} || {Key, Value} <- DataIn ],
 
-    % loginattr was used to find this record, so we know it must exist
-    [UserName0] = proplists:get_value(LoginAttr, Data),
-    UserName1 = string:to_lower(UserName0),
-    UserName2 = re:replace(UserName1, "[^0-z0-9_-]", "_", [{return, list}, global]),
-    UserName = characters_to_binary(UserName2),
+    % Since we just downcased the entire response, we need to downcase
+    % the LoginAttr to ensure it doesn't blow up and throw an exception
+    LCLoginAttr = string:to_lower(LoginAttr),
+
+    % loginattr was used to find this record, so we know it must exist;
+    % however, multiple LoginAttr fields may exist in the LDAP record, take
+    % the first
+    [CanonicalUserName|_] = [ canonical_username(U) || U <- proplists:get_value(LCLoginAttr, Data) ],
```

Comment:
I think we will want `CanonicalUserName = canonical_username(value_of(...))`, but yes, will update on next push, thanks!